lib/service/connect.go

```diff
@@ -151,7 +148,7 @@ func (process *TeleportProcess) authServerTooOld(resp *proto.PingResponse) error
 			process.log.Warnf("Teleport instance is too new. This instance is running v%d. The auth server is running v%d and only supports instances on v%d or v%d.", teleportVersion.Major, serverVersion.Major, serverVersion.Major, serverVersion.Major-1)
 			return nil
 		}
-		return trace.NotImplemented("Teleport instance is too new. This instance is running v%d. The auth server is running v%d and only supports instances on v%d or v%d. To connect anyway pass the --skip-version-check flag.", teleportVersion.Major, serverVersion.Major, serverVersion.Major, serverVersion.Major-1)
+		return trace.Wrap(invalidVersionErr{err: fmt.Errorf("Teleport instance is too new. This instance is running v%d. The auth server is running v%d and only supports instances on v%d or v%d. To connect anyway pass the --skip-version-check flag.", teleportVersion.Major, serverVersion.Major, serverVersion.Major, serverVersion.Major-1)})
```

Comment:
We could be cool and define `invalidVersionErr` as containing `teleportVersion.Major` and `serverVersion.Major` here, and move the message to its `Error()`.